tests/dialects/test_hive.py

```diff
@@ -152,7 +152,7 @@ def test_ddl(self):
                 "duckdb": "CREATE TABLE x (w TEXT)",  # Partition columns should exist in table
                 "presto": "CREATE TABLE x (w VARCHAR, y INTEGER, z INTEGER) WITH (PARTITIONED_BY=ARRAY['y', 'z'])",
                 "hive": "CREATE TABLE x (w STRING) PARTITIONED BY (y INT, z INT)",
-                "spark": "CREATE TABLE x (w STRING) PARTITIONED BY (y INT, z INT)",
+                "spark": "CREATE TABLE x (w STRING, y INT, z INT) PARTITIONED BY (y, z)",
```

Comment:
what about for spark 2? i feel like this was done on purpose?

Reply:
I don't know wtf is going on here. Documentation is horrible, of course.
spark 2 seems to use hive format.
spark 3 seems to introduce a new format, in addition to the old format.
`CREATE TABLE ... USING ICEBERG...` seems to be the new format. And the code was already forcing that for spark 2 and 3. So I bet there are many cases that were never working...

We only just caught this at airbnb because the update the converts all vars to uppercase. We were setting the kind to lowercase "table", so this check was never true: https://github.com/tobymao/sqlglot/blob/main/sqlglot/dialects/dialect.py#L664